lib/Zonemaster/Backend/RPCAPI.pm

```diff
@@ -208,97 +251,105 @@ sub _check_domain {
-    }
+        if ( defined $syntax_input->{profile} ) {
+            my @profiles = map lc, Zonemaster::Backend::Config->load_config()->ListPublicProfiles();
+            return { status => 'nok', message => encode_entities( "Invalid profile option format" ) }
+            unless ( grep { $_ eq lc $syntax_input->{profile} } @profiles );
+        }
 
-    my ( $dn, $dn_syntax ) = $self->_check_domain( $syntax_input->{domain}, 'Domain name' );
+        my ( $dn, $dn_syntax ) = $self->_check_domain( $syntax_input->{domain}, 'Domain name' );
 
-    return $dn_syntax if ( $dn_syntax->{status} eq 'nok' );
+        return $dn_syntax if ( $dn_syntax->{status} eq 'nok' );
 
-    if ( defined $syntax_input->{nameservers} && @{ $syntax_input->{nameservers} } ) {
-        foreach my $ns_ip ( @{ $syntax_input->{nameservers} } ) {
-            my ( $ns, $ns_syntax ) = $self->_check_domain( $ns_ip->{ns}, "NS [$ns_ip->{ns}]" );
-            return $ns_syntax if ( $ns_syntax->{status} eq 'nok' );
-        }
+        if ( defined $syntax_input->{nameservers} && @{ $syntax_input->{nameservers} } ) {
+            foreach my $ns_ip ( @{ $syntax_input->{nameservers} } ) {
+                my ( $ns, $ns_syntax ) = $self->_check_domain( $ns_ip->{ns}, "NS [$ns_ip->{ns}]" );
+                return $ns_syntax if ( $ns_syntax->{status} eq 'nok' );
+            }
 
-        foreach my $ns_ip ( @{ $syntax_input->{nameservers} } ) {
-            return { status => 'nok', message => encode_entities( "Invalid IP address: [$ns_ip->{ip}]" ) }
-                unless( !$ns_ip->{ip} || $ns_ip->{ip} =~ /^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$/ || $ns_ip->{ip} =~ /^([0-9A-Fa-f]{1,4}:[0-9A-Fa-f:]{1,}(:[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})?)|([0-9A-Fa-f]{1,4}::[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})$/);
+            foreach my $ns_ip ( @{ $syntax_input->{nameservers} } ) {
+                return { status => 'nok', message => encode_entities( "Invalid IP address: [$ns_ip->{ip}]" ) }
+                    unless( !$ns_ip->{ip} || $ns_ip->{ip} =~ /^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$/ || $ns_ip->{ip} =~ /^([0-9A-Fa-f]{1,4}:[0-9A-Fa-f:]{1,}(:[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})?)|([0-9A-Fa-f]{1,4}::[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3})$/);
 
-            return { status => 'nok', message => encode_entities( "Invalid IP address: [$ns_ip->{ip}]" ) }
-              unless ( !$ns_ip->{ip}
-                || Zonemaster::Engine::Net::IP::ip_is_ipv4( $ns_ip->{ip} )
-                || Zonemaster::Engine::Net::IP::ip_is_ipv6( $ns_ip->{ip} ) );
-        }
+                return { status => 'nok', message => encode_entities( "Invalid IP address: [$ns_ip->{ip}]" ) }
+                unless ( !$ns_ip->{ip}
+                    || Zonemaster::Engine::Net::IP::ip_is_ipv4( $ns_ip->{ip} )
+                    || Zonemaster::Engine::Net::IP::ip_is_ipv6( $ns_ip->{ip} ) );
```

Comment:
Please add a comment that the two tests of the provided IP address are complementary, but not the same. Both are needed.

Reply:
Fixed in dcf35a6